### ISSUE TITLE
Can (e.g.) footnote-dw include thiscite when there is a post-postnote?

### ISSUE THREAD

**Issue by pripple** [plk/biblatex#1076]
Hi, I would like to further investigate on the point made in https://github.com/plk/biblatex/issues/918#issuecomment-636011886.

I am using the `footnote-dw`-style and I am not sure how this works with other styles, but with this style, it can output the localization string for `thiscite` before a page number or page sequence in the postnote, when the preceding full citation information already contains a page rage as is common with journal articles or texts in edited collections. I have tried several ways to have BibLaTeX add that string when I am including a post-postnote, but so far with no luck. Maybe there is another way I am supposed to do this? This is what I have tried so far:

```latex
% !TEX encoding = UTF-8 Unicode
\documentclass{article}
\usepackage[ngerman]{babel}
\usepackage{csquotes}
\usepackage[style=footnote-dw]{biblatex}\usepackage{filecontents}
\begin{filecontents}{\jobname.bib}
@article{my_article,
	author = {Doe, John},
	journal = {The \LaTeX{} Journal},
	number = {3},
	pages = {393–396},
	title = {A Bibliographer’s \TeX nic Inquiry},
	volume = {25},
	year = {1989}}
\end{filecontents}
\addbibresource{\jobname.bib}
\setlength\parindent{0pt}
\begin{document}
	
Als erstes einfach eine Seite aus dem Artikel.\fullcite[Vgl.][394]{my_article} Und noch eine andere.\fullcite[Vgl.][396]{my_article} 

In those two examples, you can see that the localization string for \texttt{thiscite}, „hier“, is put out.

Wie sieht es „bei \emph{besonderen} Zusätzen“\fullcite[394, Hervorhebung im Original]{my_article} aus?

Here, it is omitted, obviously, because the \texttt{ifpages}-test fails. Another attempt, using the command which “formats i[t]s argument […] in the same format as postnote”:

Noch ein „weiterer Versuch“\fullcite[\pnfmt{393}, Hervorhebung entfernt]{my_article} …

It \emph{does} add the pages prefix „S. “, however, it doesn’t add the localization string for \texttt{thiscite}, „hier“.

Nochmal „eine \emph{andere} Variante“\fullcite[\passifpages{\pnfmt{393}}, Hervorhebung im Original]{my_article} …

This doesn’t seem to make any difference.

\end{document}
```

In `biblatex-dw/cbx/standard-dw.cbx`, this is the section where `thiscite` is added:

```latex
\def\cbx@opt@citepages@separate{%
  \providetoggle{cbx:fullcite}%
  \renewbibmacro*{cite:citepages}{%
    \global\togglefalse{cbx:fullcite}}%
  \renewbibmacro*{cite:full:citepages}{%
    \global\toggletrue{cbx:fullcite}}%
  \renewbibmacro*{cite:postnote}{%
    \ifboolexpr{
    togl {cbx:fullcite}
    and
    test {\iffieldpages{postnote}}
    and
    test {\ifnumequal{\value{citecount}}{\value{citetotal}}}
    and
    ( not test {\iffieldundef{pages}}
      or
      ( not test {\iffieldundef{pagetotal}}
        and
        bool {bbx:pagetotal}
      ) 
    )
    }
      {\usebibmacro{cite:postnote:pages}}
      {\usebibmacro{cite:postnote:ibidpage}}}
  \providebibmacro*{cite:postnote:pages}{%
    \setunit{\postnotedelim}%
    \bibstring{thiscite}%      <--------------------
    \setunit{\addspace}%
    \printfield{postnote}}}
```

I’m not fully familiar with how all of this works, but I would have thought this works together with this line in `biblatex/biblatex.sty`:

```latex
  \protected\def\pnfmt{\printtext[postnote]}
```

I would appreciate most a solution where either

- `pnfmt` included `thiscite` or, even better,
- instead of just evaluating if the postnote `ispages`, trying to split off the leading part of the postnote as far as it `ispages`, then just adding the rest.

Thank you, take care –

**Comment by moewew** [comment]
You'll want to use `\passifpages` on the text that would normally fail the `\ifpages` check. So you don't need to wrap `\pnfmt`, you need to wrap the `, Hervorhebung im Original`.

```latex
\documentclass{article}
\usepackage[ngerman]{babel}
\usepackage{csquotes}
\usepackage[style=footnote-dw]{biblatex}

\makeatletter
\let\passifpages\@firstofone
\PagesCheckSetup{\let\passifpages\@gobble}
\makeatother

\begin{filecontents}{\jobname.bib}
@article{my_article,
  author  = {Doe, John},
  journal = {The \LaTeX{} Journal},
  number  = {3},
  pages   = {393–396},
  title   = {A Bibliographer’s \TeX nic Inquiry},
  volume  = {25},
  year    = {1989},
}
\end{filecontents}
\addbibresource{\jobname.bib}

\setlength\parindent{0pt}

\begin{document}
	
Als erstes einfach eine Seite aus dem Artikel.\fullcite[Vgl.][394]{my_article}
Und noch eine andere.\fullcite[Vgl.][396]{my_article} 

In those two examples, you can see that the localization string for \texttt{thiscite}, „hier“, is put out.

Nochmal „\emph{so} gehts“\fullcite[\pnfmt{393}\passifpages{, Hervorhebung im Original}]{my_article} …

Nochmal „eine \emph{andere} Variante“\fullcite[\passifpages{\pnfmt{393}, Hervorhebung im Original}]{my_article} …
\end{document}
```

**Comment by pripple** [comment]
> You'll want to use `\passifpages` on the text that would normally fail the `\ifpages` check. So you don't need to wrap `\pnfmt`, you need to wrap the `, Hervorhebung im Original`.

Oh, okay, thank you, I tried to copy from the other thread, however, it seems I have picked up the example for how _not_ to do it. Sorry!

> ```latex
> \makeatletter
> \let\passifpages\@firstofone
> \PagesCheckSetup{\let\passifpages\@gobble}
> \makeatother
> ```

Still, this is the important part – without this, it doesn’t work. I wasn’t aware I had to include that in the preamble – it didn’t seem to throw any errors …

At least with the styles I use, it always displays the page, even when it is exactly identical (which is not ideal, but I wont investigate that further now … At least, that way I am not running in the issue Leo (zaeph) was talking about.

Thank you!
